app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -332,7 +408,13 @@ static Long timeForNextRead() {
             // A new packet should arrive but chance is we have missed it...
             return 30000L;
         }
-        return (DEXCOM_PERIOD - gapTime) + 2000;
+
```

Comment:
I think because of where timeForNextRead() is called, the choice is either to be 10 later picking up from local lan wixels when http mode is also used or guaranteeing that we will have to go to the network twice when using the parakeet http service at all. I think this is a good tradeoff. I am not quite sure of a better easy optimisation which would improve the performance.